bin/download_release_dependencies.py

```
@@ -0,0 +1,164 @@
+    print(f"Downloading {url} to {dest}...")
+    headers = {"User-Agent": "Mozilla/5.0"}
+    req = urllib.request.Request(url, headers=headers)
+    with urllib.request.urlopen(req) as response, open(dest, "wb") as out_file:
+        shutil.copyfileobj(response, out_file)
+    print("Download completed.")
+
+
+def extract_zip(zip_path: Path, extract_to: Path) -> None:
+    print(f"Extracting {zip_path}...")
+    with zipfile.ZipFile(zip_path, "r") as zip_ref:
+        zip_ref.extractall(extract_to)
```

Ruff (0.15.20)

[error] Uses of `tarfile.extractall()`

(S202)